modules/drivers/lidar/lidar_robosense/include/component/component_driver.h

```diff
@@ -0,0 +1,98 @@
+/******************************************************************************
+ * Copyright 2021 The Apollo Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *****************************************************************************/
+
+#pragma once
+#include <memory>
+#include <string>
+#include <thread>
+
+#include "cyber/cyber.h"
+#include "modules/drivers/lidar/lidar_robosense/include/driver/driver.h"
+#include "modules/drivers/lidar/lidar_robosense/proto/sensor_suteng.pb.h"
+
+namespace apollo {
+namespace drivers {
+namespace robosense {
+using apollo::cyber::Component;
+class CompRoboDriver : public Component<> {
+ public:
+  ~CompRoboDriver() {
+    if (device_thread_->joinable()) {
+      device_thread_->join();
+    }
+  }
+  bool Init() {
+    // read config file
+    apollo::drivers::suteng::SutengConfig config;
+    if (!apollo::cyber::common::GetProtoFromFile(config_file_path_, &config)) {
+      AERROR << "Failed to load config file";
+      return false;
+    }
+    AINFO << "config:" << config.DebugString();
+
+    // set default main frame
+    if (!config.has_main_frame() && config.frame_id() == "robosense16_back") {
+      config.set_main_frame(true);
+    }
+
+    RobosenseDriver *driver =
+        robosense::RobosenseDriverFactory::create_driver(config);
+    if (driver == nullptr) {
+      return false;
+    }
+    writer_ = node_->CreateWriter<apollo::drivers::suteng::SutengScan>(
+        config.scan_channel());
+
+    dvr_.reset(driver);
+    dvr_->init();
+    // spawn device poll thread
+    runing_ = true;
+    device_thread_ = std::shared_ptr<std::thread>(
+        new std::thread(std::bind(&CompRoboDriver::device_poll, this)));
+    device_thread_->detach();
```

Comment:
By default, a detached thread is not joinable. I personally not very clear the life cyber of a detached thread w.r.t. the main thread.